CRM/Volunteer/BAO/Assignment.php

```diff
@@ -311,6 +313,12 @@ public static function createVolunteerActivity(array $params) {
       }
     }
 
+    // Saving the Volunteer role custom field for the activity.
```

Comment:
The following block should be unnecessary. The `foreach` loop immediately above this automagically converts `$params['volunteer_role_id']` to `$params["custom_$n"]`, and `$params` already takes `$defaults` into account a few lines before that. What kind of problems did you run into that you needed to add this code?